fuel-core-interfaces/src/model/da_block_height.rs

```diff
@@ -0,0 +1,85 @@
+use derive_more::{
+    Add,
+    Deref,
+    Display,
+    From,
+    Into,
+};
+use std::ops::{
+    Add,
+    Rem,
+    Sub,
+};
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    Default,
+    PartialEq,
+    PartialOrd,
+    Eq,
+    Add,
+    Ord,
+    Display,
+    Into,
+    From,
+    Deref,
+    Hash,
+)]
+pub struct DaBlockHeight(pub u64);
+
+impl From<DaBlockHeight> for Vec<u8> {
+    fn from(height: DaBlockHeight) -> Self {
+        height.0.to_be_bytes().to_vec()
+    }
+}
+
+impl From<usize> for DaBlockHeight {
+    fn from(n: usize) -> Self {
+        DaBlockHeight(n as u64)
+    }
+}
+
+impl Add<u64> for DaBlockHeight {
+    type Output = Self;
+
+    fn add(self, other: u64) -> Self::Output {
+        Self::from(self.0 + other)
+    }
+}
+
+impl Sub for DaBlockHeight {
```

Comment:
Sub and Rem should be derivable via [derive_more](https://jeltef.github.io/derive_more/derive_more/)

Reply:
Looks like Rem isn't but Sub is

Reply:
Got derived Rem working, although it required using `#[rem(forward)]` to avoid breaking other code. Here's the explanation for what `forward` does:

> A simple way of explaining the reasoning behind this difference between Add and Mul deriving, is looking at arithmetic on meters. One meter can be added to one meter, to get two meters. Also, one meter times two would be two meters, but one meter times one meter would be one square meter. As this second case clearly requires more knowledge about the meaning of the type in question deriving for this is not implemented.

> NOTE: In case you don’t want this behaviour you can add #[mul(forward)] in addition to #[derive(Mul)]. This will instead generate a Mul implementation with the same semantics as Add.

https://jeltef.github.io/derive_more/derive_more/mul.html